File: algorithms/CLRS/ch02_gettingstarted/BinarySearch.java
public class BinarySearch 
{
	public static void main(String[] args)
	{
		int[] numbers = {23, 34, 67, 78, 79, 80, 92, 95, 97, 100};
		int key = 34;

		binSearch(numbers,key);
	}	

	static void binSearch(int[] arr, int key)
	{
		int low = 0;
		int high = arr.length - 1;
		int mid;

		while(low <= high) 
		{
			mid = low + high / 2;

			if(key == arr[mid]){
				System.out.println("The key found at position " + mid);
				break;
			}
			else if(key > arr[mid])
				low = mid + 1;
			else if(key < arr[mid])
				high = mid - 1;
			else
				System.out.println("The key is not present in the array.");
		}
	}
}

File: c++/udemy_Abdul_Bari/sec_08_arrays/binary_search.cpp
#include <iostream>

using namespace std;

int main() 
{
    int A[10], n = 10;

    cout << "Enter numbers: ";
    
    for (int i = 0; i < n; ++i)
        cin >> A[i];

    cout << "Enter the number to be searched: ";
    int key;
    cin >> key;
    int low = 1, high = 10;
    int mid;
    int step = 1;

    bool found = false;
    while (low < high)
    {
        mid = (low + high) / 2;
        cout << "key: " << key << " ";
        cout << "step: " << step++ << ", low: " << low << ", high: "
            << high << ", mid << " << mid << endl;
        if (key < A[mid])
        {
            high = mid - 1;
        }
        else if (key > A[mid])
        {
            low = mid + 1;
        }
        else if (key == A[mid]) {
            found = true;
            break;
        }
    }

    if (found)
        cout << "found\n";
    else
        cout << "not found\n";

    return 0;
}